### PR TITLE
wip: Feat/ini web

### PR DESCRIPTION
**Pull request by shunkakinoki**
Todos (from tg)

- [x] The common package should pull in from workspaces:* rather than using a version number
- [x] ~~I believe addressToProjectivePoint is misnamed. It’s just using the uncompressed byte representation of the public key and mapping it to a projective point. You could use ProjectivePoint.fromHex() to do this as well!~~
- [x] ~~To clarify the address, the only time we should be thinking about the address is when the signer connects their wallet. Otherwise, we should think of the nullifier as representing the signer~~
- [x] In the scripts (like `generateContracts`) I notice you’ve add the `yarn` command at the beginning. Instead, these scripts should be called from the package.json scripts which allows us to skip the yarn command
- [x] Some console logs tidying that we can take out as well 🙂

### CODE REVIEW

File: web/src/utils/generateInputs.ts [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+}
+
+interface SignerWithTree extends Signer {
+  siblings: number[]
+  pathIndices: any[]
+}
+
+const isDebugMode = false;
+
+function assert(condition: boolean, message?: string) {
```
Comment (bankisan) [inline web/src/utils/generateInputs.ts]: We can remove the `assert()` calls from the input generation. Was helpful for testing in the `circuits/` code, here we would want to respond with relevant errors

File: web/src/utils/generateInputs.ts [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+  if (isDebugMode && !condition) {
+    throw new Error(message || 'Assertion failed');
+  }
+}
+
+export const generateInputs = async (
```
Comment (bankisan) [inline web/src/utils/generateInputs.ts:40]: At some point we should extract this out as a common utility for testing and the client, but this is good for now!

File: common/dist/lib/signature-proof.js [hunk 3]
```diff
@@ -76,3 +78,39 @@ export const executeTransactionData = (tx) => {
...
+    }
+    return result;
+};
+export const publicKeyToAddress = (publicKey) => {
+    const pub = secp256k1.ProjectivePoint.fromHex(publicKey).toRawBytes(false);
+    console.log(pub);
+    const addr = "0x" + bytesToHex(keccak_256(pub.subarray(1, 65))).slice(24);
+    return addr;
+};
+export const addressToProjectivePoint = (address) => {
```
Comment (bankisan) [inline common/dist/lib/signature-proof.js:98]: We can remove the `addressToProjectivePoint` and `powMod` helper above

File: circuits/scripts/generateTestFixtures.ts [hunk 4]
```diff
@@ -106,6 +106,8 @@ const createFixture = async (
     `0xFEfC6BAF87cF3684058D62Da40Ff3A795946Ab06`,
     31337n,
   )
+  console.log(hashed)
+  console.log(hashed.length)
```
Comment (bankisan) [inline circuits/scripts/generateTestFixtures.ts]: cleanup 🧹

File: web/src/utils/cn.ts [hunk 5]
```diff
@@ -0,0 +1,6 @@
+import { ClassValue, clsx } from "clsx"
```
Comment (bankisan) [inline web/src/utils/cn.ts]: Could we rename this file to `classname` or something descriptive? I don't mind the function being short, but it's hard to grok why we're importing it with a short name

**Review by bankisan — CHANGES_REQUESTED** [review]

File: web/src/utils/cn.ts (hunk 5 above)
Comment (shunkakinoki) [reply web/src/utils/cn.ts]: Gotit! Definitely haha

File: web/src/utils/generateInputs.ts (hunk 1 above)
Comment (shunkakinoki) [reply web/src/utils/generateInputs.ts]: Indeed removing haha

File: web/src/utils/generateInputs.ts (hunk 2 above)
Comment (shunkakinoki) [reply web/src/utils/generateInputs.ts:40]: True! Completely with you.

**Pull request merged**